Edit in $EDITOR

let-def opened the pull request:
This PR allows to manipulate the log in the text editor set by user ($EDITOR or whatever xdg-open or open pick).
To be merged after #9

let-def committed:
Lwd_table: add iter function

let-def committed:
minor refactoring

let-def committed:
implement log manipulation in $EDITOR

let-def committed:
update TODO

craigfe commented:
I really like this improvement. 

One thought: the `` `Right `Activate`` action fails to trigger if the logs have not yet been loaded in the pane; it might be nice if the command to open the editor was buffered so that the logs will be opened as soon as they are ready. While experimenting with this branch, I often run into this issue of needing to press `l` more than once to get what I want :slightly_smiling_face:

Julow commented on src/main.ml:
```diff
@@ -1,488 +1,323 @@
+                | Some x -> [ x ]
+                | None -> [] )
+          in
+          let candidates = candidates @ [ "xdg-open"; "open" ] in
```
`$VISUAL` should be checked before `$EDITOR`, it should be a terminal editor too.

On macos, `open` terminates immediately, sometimes before the file is loaded by the editor.

Julow commented on src/main.ml:
```diff
@@ -1,488 +1,323 @@
+               (fun bin ->
+                 Sys.command (Filename.quote bin ^ " " ^ safe_name) <> 127)
+               candidates)
+        with _ -> Sys.remove fname );
```
Is this reachable ?

let-def committed:
open editor as soon as log is received

let-def commented:
I adressed both of your concerns:
- `VISUAL` is checked before `EDITOR`
- if the log has not been received yet, the command is scheduled
- now a single log file is created (and rewritten as necessary) and deleted when the software is closed
